pkg/scheduler/scheduler.go

```diff
@@ -96,20 +96,42 @@ func (s *Scheduler) Run(request *types.ContainerRequest) error {
 	go s.schedulerMetrics.CounterIncContainerRequested(request)
 	go s.eventRepo.PushContainerRequestedEvent(request)
 
-	err = s.containerRepo.SetContainerState(request.ContainerId, &types.ContainerState{
-		ContainerId: request.ContainerId,
-		StubId:      request.StubId,
-		Status:      types.ContainerStatusPending,
-		WorkspaceId: request.WorkspaceId,
-		ScheduledAt: time.Now().Unix(),
-	})
+	quota, err := s.getConcurrencyLimit(request)
+	if err != nil {
+		return err
+	}
+
+	err = s.containerRepo.SetContainerStateWithConcurrencyLimit(quota, request)
```

Comment:
we shouldn't need a separate state

Reply:
Not a separate state, it's the same state